pkg/kubelego/watch.go

```diff
@@ -27,13 +27,25 @@ func ingressWatchFunc(c *kubernetes.Clientset, ns string) func(options k8sMeta.L
 	}
 }
 
-func (kl *KubeLego) requestReconfigure() {
-	kl.workQueue.Add(true)
+// requestReconfigure will trigger a resync of *all* ingress resources.
+func (kl *KubeLego) requestReconfigure() error {
+	allIng, err := ingress.All(kl)
+	if err != nil {
+		return err
+	}
+	for _, ing := range allIng {
+		key, err := cache.MetaNamespaceKeyFunc(ing.Object)
+		if err != nil {
+			return err
+		}
+		kl.workQueue.AddRateLimited(key)
+	}
+	return nil
 }
 
 func (kl *KubeLego) WatchReconfigure() {
 
-	kl.workQueue = workqueue.New()
+	kl.workQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Minute*10, time.Hour*24), "kube-lego")
```

Comment:
10 minutes seems rather a long `basedelay`.
This means that kube-lego will send an API request to LetsEncrypt at most, once every ten minutes, right?
I'd have thought 1 API req/sec would be a reasonable base.
Maybe add a note to the docs explaining the new behaviour.
And do LE have any server side limits? 
If so, it'd cool if our LE client could understand HTTP 429 headers, does it already  do that?

Reply:
So kube-lego attempts the entire validation flow in a single sync iteration, so each sync iteration can take up to 15 minutes at a time. We have a base back off of 10 minutes, which means after a single sync has failed, we'll wait 10 minutes before processing that item again.

Reply:
Re 429 headers - these are sadly not yet returned 😢